src/Rings/mpoly-localizations.jl

```diff
@@ -533,6 +533,13 @@ function ==(T::AbsMPolyMultSet, U::AbsMPolyMultSet)
   return (issubset(T, U) && issubset(U, T))
 end
 
+function Base.hash(T::AbsMPolyMultSet, h::UInt)
+  b = 0x7ce51a28c47ec5e1 % UInt
+  h = hash(typeof(T), h)
+  h = hash(ambient_ring(T), h)
+  return xor(h, b)
+end
```

Comment:
Thanks for the effort! However, this kind of hashing will not be sufficient. Instead, I propose to throw an error here which requires the user to overwrite the hash function for their concrete type. 

Can I do that and push to this PR?

Reply:
Do you mean something similar to this?
 https://github.com/oscar-system/Oscar.jl/blob/96c2eed7c8b4ed2289290ffc0254b7ddf5122d12/src/Rings/mpoly-localizations.jl#L529

Yeah, that would be great. I am off now for the next ~ 2 hours. So if you push it in the meantime, we will not have a conflict. If thats too spontaneous for you, please text me on slack to coordinate.

Reply:
I added a warning in 7c336c38abbed2e8463a2db8add5adf6f4dc0ab4. Is this fine with you?

Reply:
After looking a bit deeper into the concrete subtypes, I think there cannot be special implementations of `hash` for each type. There are many different implementations for `issubset` for different pairs of types, and equality is just defined generically using `issubset`. Since AbsMPolyMultSets of different concrete types can be equal, their hashes should coincide as well. And I do not see how to do that with specialized implementations.